engine/executor/agg_func_prom.go

```diff
@@ -181,3 +183,95 @@ func almostEqual(a, b, epsilon float64) bool {
 	}
 	return diff/math.Min(absSum, math.MaxFloat64) < epsilon
 }
+
+type CallFn func([]int64, []int64, []float64, []float64, int64, *influxql.PromSubCall) (float64, bool)
+
+var promSubqueryFunc map[string]CallFn = make(map[string]CallFn)
+
+func init() {
+	promSubqueryFunc["rate_prom"] = rate
+}
+
+func CalcReduceResult(prevT, currT []int64, prevV, currV []float64, isCounter bool) (int64, int64, float64, float64, float64) {
+	var firstTime, lastTime int64
+	var firstValue, lastValue float64
+	if len(prevT) > 0 {
```

Comment:
adjust conditional logic to reduce conditional nesting，Is this modification reasonable：

```go
if len(prevT) == 0 && len(currT) == 0 {
		// do something
	}

	firstTime, firstValue := prevT[0], prevV[0]
	if len(prevT) == 0 {
		firstTime, firstValue = currT[0], currV[0]
	}

	lastTime, lastValue := currT[len(currT)-1], currV[len(currV)-1]
	if len(currT) == 0 {
		lastTime, lastValue = prevT[len(prevT)-1], prevV[len(prevV)-1]
	}

	reduceResult := lastValue - firstValue
```

Reply:
The original code can reduce the number of judgments.